src/pu-main.c

```diff
@@ -220,13 +363,30 @@ static GOptionEntry option_entries_version[] = {
     { NULL }
 };
 
+static GOptionEntry option_entries_validate[] = {
+    { "skip-checksums", 's', G_OPTION_FLAG_NONE, G_OPTION_ARG_NONE,
+        &arg_validate_skip_checksums, "Skip checksum verification for all input files", NULL },
+    { "size", 0, G_OPTION_FLAG_NONE, G_OPTION_ARG_STRING,
+        &arg_validate_file_size, "Size of loop device. Default is 1G", "SIZE" },
+    { "write-file", 'w', G_OPTION_FLAG_NONE, G_OPTION_ARG_NONE,
+        &arg_validate_write_file, "Install the package to a file as an additional validation step",
+        NULL },
+    { "keep-file-as", 'k', G_OPTION_FLAG_NONE, G_OPTION_ARG_FILENAME,
+        &arg_validate_keep_file, "Keep the file. Implies --write-file", "FILENAME" },
+    { G_OPTION_REMAINING, 0, G_OPTION_FLAG_NONE, G_OPTION_ARG_STRING_ARRAY,
+        &arg_remaining, NULL, "validate PACKAGE" },
```

Comment:
I think we should *optionally* allow validating a package against a real device. In case no device was specified, a loop device is probably a good idea still.

Reply:
Added that feature with the `--device` option.